reps/2023-10-13-accelerator-support.md

```diff
@@ -0,0 +1,374 @@
+      """
+
+  @staticmethod
+  def get_ec2_instance_num_accelerators(
+      instance_type: str, instances: dict
+  ) -> Optional[int]:
+      """Get the number of accelerators of this family on ec2 instance with given type.
+
+      Args:
+          instance_type: The ec2 instance type.
+          instances: Map from ec2 instance type to instance metadata returned by
+              ec2 `describe-instance-types`.
+
+      Returns:
+          The number of accelerators of this family on the ec2 instance
+          with given type.
+          Return None if it's unknown.
+      """
+      return None
+
+  @staticmethod
+  def get_ec2_instance_accelerator_type(
+      instance_type: str, instances: dict
+  ) -> Optional[str]:
+      """Get the accelerator type of this family on ec2 instance with given type.
+
+      Args:
+          instance_type: The ec2 instance type.
+          instances: Map from ec2 instance type to instance metadata returned by
+              ec2 `describe-instance-types`.
+
+      Returns:
+          The accelerator type of this family on the ec2 instance with given type.
+          Return None if it's unknown.
+      """
+      return None
+```
+
+Besides defining the accelerator subclass, the accelerator resource needs to be marked as unit instance resource in config `RAY_custom_unit_instance_resources`.
+
+### Ray Train
+
+Ray Train will provide 2 new APIs to support different accelerator types. 
+
+
+#### 1. Specify Accelerator Type in `ScalingConfig`
+
+Ray Core automatically detects the accelerator type, and treats it as an additional resource tag. Ray Train will therefore be able to launch training jobs on specific accelerator types. We add a new initialization argument `accelerator_type` for `ray.train.ScalingConfig`.
+
+
+```python
+# CPU
+ScalingConfig(num_workers=4)
+
+# GPU
+# If no accelerator is specified, schedule workers on any GPU node
+ScalingConfig(num_workers=4, use_gpu=True)
+
+# GPU (A100)
+# If specified, schedule workers only on A100 nodes
+ScalingConfig(num_workers=4, use_gpu=True, accelerator_type="A100")
+
+# AWS Trainium
+ScalingConfig(num_workers=4, resources_per_worker={"neuron_cores": 1})
+
+# TPU (TPU-v3)
+ScalingConfig(num_workers=4, accelerator_type="TPU-V3", resources_per_worker={"TPU": 1})
+
+# HPU (Gaudi2)
+ScalingConfig(num_workers=4, accelerator_type="Gaudi2", resources_per_worker={"HPU": 1})
+```
+
+Internally, Ray Train adds the `accelerate_type` parameter in `ray.remote()` to start the training workers.
+
+#### 2. Specify PyTorch Backend for each Accelerator Family
+
+Different accelerator family uses different backend to launch deep learning training. Ray Train will implement a different backend class for each accelerator family to launch distributed groups and set environment variables.
+
+| Accelerator | Communication Backend | Developer APIs | User APIs | 
+| - | - | - | - |
+| GPU      | `nccl`, `gloo` | TorchBackend | `TorchConfig()` |
```

Comment:
@woshiyyya Can you add an example of Intel GPU

Reply:
Done.